nixos/modules/services/web-apps/comfyui.nix

```diff
@@ -0,0 +1,498 @@
+        description = "ComfyUI base package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        example = "yourUser";
+        description = ''
+          The user to run ComfyUI as.
+          By default, a user named `${defaultUser}` will be created whose home
+          directory will contain input, output, custom nodes and models.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = defaultGroup;
+        example = "yourGroup";
+        description = ''
+          The group to run ComfyUI as.
+          By default, a group named `${defaultUser}` will be created.
+        '';
+      };
+
+      useCPU = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Uses the CPU for everything. Very slow, but needed if there is no hardware acceleration.
+        '';
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8188;
+        description = "Set the listen port for the Web UI and API.";
+      };
+
+      customNodes = mkOption {
+        type = types.listOf types.package;
+        default = [];
+        description = "custom nodes to add to the ComfyUI setup. Expects a list of packages from pkgs.comfyui-custom-nodes";
+      };
+
+      listen = mkOption {
+        type = types.str;
+        # Assume a higher security posture by default.
+        default = "127.0.0.1";
+        description = "The net mask to listen to.";
+        example = "0.0.0.0";
+      };
+
+      cors-origin-domain = mkOption {
+        type = types.str;
+        default = "disabled";
+        description = ''The CORS domain to bless.  Use "disabled" to disable.  This must include a port'';
+        example = "foo.com:443";
+      };
+
+      cuda-malloc = mkOption {
+        type = types.bool;
+        default = cfg.cudaSupport;
+      };
+
+      max-upload-size = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+      };
+
+      multi-user = mkOption {
+        type = types.bool;
+        default = false;
+      };
+
+      cuda-device = mkOption {
+        type = types.nullOr types.str;
+        description = ''The CUDA device to use.  Query for this using lspci or lshw.  Leave as null to auto-detect and/or use Nix CUDA settings (verify this before merging!).'';
+        default = null;
+      };
+
+      verbose = mkOption {
+        type = types.bool;
+        description = ''Use verbose logging.'';
+        default = false;
+      };
+
+      cross-attention = mkOption {
+        # TODO: Learn2enum in Nix.
+        # Valid types should be "split", "quad", and "pytorch".
+        type = types.nullOr types.enum;
+        # TODO: Learn what cross-attention is, and describe it here.
+        description = '' '';
+        default = null;
+      };
+
+      preview-method = mkOption {
+        type = types.nullOr types.oneOf [
```

Comment:
This should probably be an enum and put in parentheses. Could be the cause of your manual build error.

i.e. `types.nullOr (types.enum [...])`

Reply:
... wow this is what fixed it for me!  Thank you!!!  Once I had this sorted, it was just adding some `description` fields as needed.